### ISSUE TITLE
2086 do not warn if contains a *

### ISSUE THREAD

**Issue by kkmuffme** [koalaman/shellcheck#1437]
#### For bugs
- Rule Id (if any, e.g. SC1000): 2086
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
rm -rf ${dir}/results/*
```

#### Here's what shellcheck currently says:
Double quote to prevent globbing and word splitting

#### Here's what I wanted or expected to see:
If a rm command contains a *, do not throw this message since in 99.9% of cases it is a false positive (and if it is actually intended by the dev to delete something from a path with a *, then he will already have put quotes as otherwise his script wouldn't work in the first place)

**Comment by matthewpersico** [comment]
Look again at the comment - it's asking you to quote the `${dir}`, not the whole path. Try this on shellcheck.net:

```
#!/bin/bash

dir=$1
rm -rf ${dir}/results/*
rm -rf "${dir}"/results/*
```

**Comment by ngzhian** [comment]
We do have end location markers, and it shows up on a master build, v0.6.0, but not v0.5.0. Maybe shellcheck.net is behind?

```
In test.sh line 4:
rm -rf ${dir}/results/*                                                                                            
       ^----^ SC2086: Double quote to prevent globbing and word splitting.
```

**Comment by orbea** [comment]
Note that quoting only part of the string is not fully portable, but I suppose this is out of scope for shellcheck?

**Comment by koalaman** [comment]
ShellCheck.net's "console output" is actually just HTML generated by JavaScript based on the JSON output. It only pretends to look like the console version. Showing end positions and fixes is not yet implemented there.

I definitely assumed `rm -rf "${dir}"/results/*` should be fine. @orbea, which shells fail to handle this?

**Comment by orbea** [comment]
Sorry, I don't recall exactly and may be conflating memories since I can't reproduce any issues at the moment, sorry for the noise.

**Comment by koalaman** [comment]
Sounds good, then.

ShellCheck is technically correct, though I see why someone might try to quote the whole word if they don't read through the admittedly long wiki page.

It shouldn't be a problem for long though, since the next version of ShellCheck (and ShellCheck.net as of today) will show `rm -rf "${dir}"/results/*` explicitly.
